### ISSUE TITLE
Network samples sometimes crash right after startup

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1590]
**_Reported by John Andersen:_**

Occasionally I see the network samples, TCP and UDP (echo-client and echo-server) crash on startup.

Commit: 161a863da715019b529cf76c79f1cdade3f98433

This was echo_client. I had modified the config slightly to turn up logging and turn off TCP and IPv4.

```
CONFIG_NETWORKING=y
CONFIG_NET_IPV6=y
CONFIG_NET_IPV4=n
CONFIG_NET_UDP=y
CONFIG_NET_TCP=n
CONFIG_TEST_RANDOM_GENERATOR=y
CONFIG_NET_BUF_LOG=y
CONFIG_SYS_LOG_NET_BUF_LEVEL=5
CONFIG_NET_LOG=y
CONFIG_SYS_LOG_NET_LEVEL=5
CONFIG_NET_SLIP_TAP=y
CONFIG_SYS_LOG_SHOW_COLOR=y
CONFIG_INIT_STACKS=y
CONFIG_PRINTK=y
CONFIG_NET_STATISTICS=y
CONFIG_NET_NBUF_RX_COUNT=14
CONFIG_NET_NBUF_TX_COUNT=14
CONFIG_NET_NBUF_DATA_COUNT=30
CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=3
CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=2
CONFIG_NET_MAX_CONTEXTS=10

CONFIG_NET_SHELL=y

CONFIG_NET_SAMPLES_IP_ADDRESSES=y
CONFIG_NET_SAMPLES_MY_IPV6_ADDR="2001:db8::2"
CONFIG_NET_SAMPLES_PEER_IPV6_ADDR="2001:db8::1"
CONFIG_NET_SAMPLES_MY_IPV4_ADDR="192.0.2.2"
CONFIG_NET_SAMPLES_PEER_IPV4_ADDR="192.0.2.1"
```

```
[net/buf] [ERR] net_buf_alloc_debug: net_nbuf_get_reserve():331: Failed to get free buffer
[net/buf] [ERR] net_buf_alloc_debug: net_nbuf_get_reserve():331: Failed to get free buffer
[net/buf] [ERR] net_buf_alloc_debug: net_nbuf_get_reserve():331: Failed to get free buffer
[net/buf] [ERR] net_buf_alloc_debug: net_nbuf_get_reserve():331: Failed to get free buffer
[net/buf] [ERR] net_buf_alloc_debug: net_nbuf_get_reserve():331: Failed to get free buffer
[net/buf] [ERR] net_buf_alloc_debug: net_nbuf_get_reserve():331: Failed to get free buffer
[net/buf] [ERR] net_buf_alloc_debug: net_nbuf_get_reserve():331: Failed to get free buffer
[net/buf] [ERR] net_buf_alloc_debug: net_nbuf_get_reserve():331: Failed to get free buffer
[net/buf] [ERR] net_buf_alloc_debug: net_nbuf_get_reserve():331: Failed to get free buffer
qemu: fatal: Trying to execute code outside RAM or ROM at 0xaaaaaaaa

EAX=0010f608 EBX=001000a0 ECX=00000000 EDX=f000e2c3
ESI=aaaaaaaa EDI=aaaaaaaa EBP=0010f658 ESP=0000e1fc
EIP=aaaaaaaa EFL=00000406 [D----P-] CPL=0 II=0 A20=1 SMM=0 HLT=0
ES =0010 00000000 ffffffff 00cf9300 DPL=0 DS   [-WA]
CS =0008 00000000 ffffffff 00cf9b00 DPL=0 CS32 [-RA]
SS =0010 00000000 ffffffff 00cf9300 DPL=0 DS   [-WA]
DS =0010 00000000 ffffffff 00cf9300 DPL=0 DS   [-WA]
FS =0010 00000000 ffffffff 00cf9300 DPL=0 DS   [-WA]
GS =0010 00000000 ffffffff 00cf9300 DPL=0 DS   [-WA]
LDT=0000 00000000 0000ffff 00008200 DPL=0 LDT
TR =0000 00000000 0000ffff 00008b00 DPL=0 TSS32-busy
GDT=     00100080 00000017
IDT=     0010c048 000007ff
CR0=0000003f CR2=00000000 CR3=00000000 CR4=00000000
DR0=00000000 DR1=00000000 DR2=00000000 DR3=00000000
DR6=ffff0ff0 DR7=00000400
CCS=0000e200 CCD=0000e206 CCO=ADDB
EFER=0000000000000000
FCW=037f FSW=0000 [ST=0] FTW=00 MXCSR=00001f80
FPR0=0000000000000000 0000 FPR1=0000000000000000 0000
FPR2=0000000000000000 0000 FPR3=0000000000000000 0000
FPR4=0000000000000000 0000 FPR5=0000000000000000 0000
FPR6=0000000000000000 0000 FPR7=0000000000000000 0000
XMM00=00000000000000000000000000000000 XMM01=00000000000000000000000000000000
XMM02=00000000000000000000000000000000 XMM03=00000000000000000000000000000000
XMM04=00000000000000000000000000000000 XMM05=00000000000000000000000000000000
XMM06=00000000000000000000000000000000 XMM07=00000000000000000000000000000000
/home/johnsa1/z/scripts/Makefile.qemu:24: recipe for target 'run' failed
make[2]: *** [run] Aborted (core dumped)
make[2]: Leaving directory '/home/johnsa1/z/samples/net/udp/udp/outdir/qemu_x86'
Makefile:173: recipe for target 'sub-make' failed
make[1]: *** [sub-make] Error 2
make[1]: Leaving directory '/home/johnsa1/z'
/home/johnsa1/z/Makefile.inc:97: recipe for target 'run' failed
make: *** [run] Error 2
```

(Imported from Jira ZEP-1737)

**Comment by nashif** [comment]
**_by Jukka Rissanen:_**

This is fixed by commit ("drivers: slip: Fix memory corruption when receiving packet").

**Comment by nashif** [comment]
**_by Jukka Rissanen:_**

Already fixed in 1.7
